src/visualizer/sortingvisualizer/Visualizer.js

```diff
@@ -136,30 +139,11 @@ const Visualizer = () => {
     } else if (isRadixSort(visualizerAlgorithm)) {
       executeRadixSort(currentAnimation, referenceArray, stackArr, false);
     } else if (isMergeSort(visualizerAlgorithm)) {
-      let nextReferenceArray = historyArr.pop();
-      setHistoryArr(historyArr);
-      setReferenceArray(nextReferenceArray);
+      executeMergeSortBackward(historyArr, setReferenceArray);
```

Comment:
Just curious as to what is the difference between the forward and backward animation and why the former requires the nextReferenceArray and the latter does not?